app/src/main/java/de/westnordost/streetcomplete/MainActivity.java

```diff
@@ -247,6 +252,36 @@ public void onServiceDisconnected(ComponentName className)
 		{
 			questController.deleteOld();
 		}
+
+		Intent intent = getIntent();
+		Uri data = intent.getData();
+		if (Intent.ACTION_VIEW.equals(intent.getAction()) && data != null)
+		{
+			if ("geo".equals(data.getScheme()))
+			{
+				double latitude = 0;
+				double longitude = 0;
+				float zoom = -1;
+
+				String geoUriRegex = "^geo:(-?[0-9]*\\.?[0-9]+),(-?[0-9]*\\.?[0-9]+).*?(?:\\?z=([0-9]*\\.?[0-9]+))?";
+				Pattern pattern = Pattern.compile(geoUriRegex);
+				Matcher matcher = pattern.matcher(data.toString());
+				while (matcher.find())
+				{
+					latitude = Double.parseDouble(matcher.group(1));
+					longitude = Double.parseDouble(matcher.group(2));
+					if (matcher.group(3) != null) {
+						zoom = Float.valueOf(matcher.group(3));
+					}
+				}
+
+				mapFragment.setPosition(new LngLat(longitude,  latitude));
+				if (zoom != -1)
+				{
+					mapFragment.setZoom(zoom);
```

Comment:
Not sure what `zoom=.0` would do. In the worst case, some `IllegalArgumentException` (probably not). Still, I think checking for `zoom > 0` would be safer here.